test/unit/response.test.js

```diff
@@ -23,6 +24,16 @@ describe('Response', function () {
             expect(jsonified).to.have.property('cookie');
         });
 
+        it('must retain Buffer nature on stream property after new Response(response.toJSON())', function () {
+            var rawResponse = {
+                    body: 'response body',
```

Comment:
@kamalaknn Also, can you add an explicit test for 
var rawResponse = { 
stream = {
  type: 'Buffer',
  data: [ 114, 101, 115, 112, 111, 110, 115, 101, 32, 98, 111, 100, 121 ]
}
}

I know this covers this usecase, but expected a seperate one as well.